lib/track.ts

```diff
@@ -98,6 +100,10 @@ export class TrackClient {
     });
   }
 
+  trackPush(data: RequestData = {}) {
```

Comment:
A few things here:

- `data` should be more strongly typed. According to [the api docs](https://customer.io/docs/api/#operation/pushMetrics), there are only four valid parameters. A `type` or `interface` should be created for `trackPush` to only allow those four (requried) parameters
- There should be special handling if `this.trackPushRoot` is `undefined`. Probably a `MissingParamError` or a new `Error` subclass

Reply:
👍 Added the typing, can make it required but the docs don't seem to annotate them as required like it does for name in regular events.